src/game_objects/grid.cpp

```diff
@@ -182,6 +187,10 @@ Construct* Grid::build(ConstructType type, int col, int row, float fx, float fy,
         children.push_back(to_add);
         candidate->set_construct(to_add);
         candidate->buildable = false;
+
+        // Traversable needed for path planning.
+        if (type == CHEST || type == ARROW_TRAP)
+            candidate->traversable = false;
```

Comment:
"Traversible" is only used for the path-finding algorithm. It won't affect how the player moves or what happens when a chest disappears. Hmm, but technically we don't need this anymore since Mimics can pass through chests and arrow_traps, so I might just remove it later if it seems to cause a problem.